src/mscorlib/shared/System/Threading/ReaderWriterLockSlim.cs

```diff
@@ -54,8 +53,10 @@ internal class ReaderWriterCount
     /// </summary>
     public class ReaderWriterLockSlim : IDisposable
     {
+        private static readonly int ProcessorCount = Environment.ProcessorCount;
```

Comment:
It's being ignored in most places currently. Maybe PlatformHelper could be extended to check affinity as well. Or if it's not expected to change, could check it once on startup. I'll leave that for a separate issue.